crates/sui-storage/src/object_store/util.rs

```diff
@@ -18,74 +21,62 @@ use tokio::time::Instant;
 use tracing::{error, warn};
 use url::Url;
 
-pub async fn get(location: &Path, from: Arc<DynObjectStore>) -> Result<Bytes, object_store::Error> {
-    let backoff = backoff::ExponentialBackoff::default();
-    let bytes = retry(backoff, || async {
-        from.get(location).await.map_err(|e| {
+pub async fn get<S: ObjectStoreGetExt>(store: &S, src: &Path) -> Result<Bytes> {
+    let bytes = retry(backoff::ExponentialBackoff::default(), || async {
+        store.get_bytes(src).await.map_err(|e| {
```

Comment:
I remember one thing that led to issues in debugging formal snapshots was in this never failing in the case where `get_bytes` was actually not a transient error. In my case, I believe the conversion to `bytes()` was failing, not the `get`. I don't recall the details, but are we sure this should always be transient? And if not, would it make sense to have a `max_retries` here?

Reply:
Discussed offline, default backoff already has a limit